### ISSUE TITLE
Add support for re-exporting crates ✨

### ISSUE THREAD

**Issue by evaspitslines** [bevyengine/bevy#9831]
I have a crate that reexports bevy :)

```Rust
use eva_voxel::ecs::system::Resource;

#[derive(Resource)]
struct MyResource {}

fn main() {
    
}
```

this code does not compile, since the Resource proc macro thinks that it should use `bevy_ecs::Resource` :( 

Could we somehow add support so it can work? 😺

**Comment by hymm** [comment]
I think you should be able to add  `use exa_voxel as bevy_ecs` as a workaround

**Comment by evaspitslines** [comment]
yes that is working :) but it would be really great to have it work out of the box 
from my understanding, the crate name is retrieved from the bevy manifest, maybe we could add a config for it to take another name 😄

**Comment by hymm** [comment]
looks like serde has an attribute for this https://serde.rs/container-attrs.html#crate

**Comment by evaspitslines** [comment]
the thing is: It would be really nice to have it work out of the box, so that I only need to write the same code as in the example ✨ . I don't know if its possible, but it would be great 😅

**Comment by nicopap** [comment]
It's an unsolved problem. with a `macro_rules` it just works if you use `$crate`, but it's not easy with proc macros.

Most crates that export a proc macro have an attribute to the proc macro that allow overwriting the import crate (like @hymm mentioned with serde)

Nothing prevents you from forking `bevy_ecs_macros` and re-export your own version of the macro that uses your crate's name as the import crate though.
